sdks/python/apache_beam/examples/complete/autocomplete_test.py

```diff
@@ -41,7 +37,7 @@ def assert_that(pcoll, matcher):
   def check_matcher(_, side_value):
     assert matcher(side_value)
     return []
-  singleton | beam.FlatMap(check_matcher, AsIter(pcoll))
+  singleton | beam.FlatMap(check_matcher, AsIter(pcoll))  # pylint: disable=expression-not-assigned
```

Comment:
This type of usage mainly exists in test code and examples. Because of the reason you pointed out. using `|` results in this expression with side effects. However we would like to prevent it in the main sdk code. I think for that reason it is better to keep it.